ci_action/library/pr_resolve.py

```diff
@@ -130,32 +122,22 @@ def read_test_annotations(
         repo_name, org = github_client.get_repo_tuple_from_github_uri(repo_uri=repo_uri)
         build_group_pr_map[f'{org.lower()}/{repo_name.lower()}'] = int(pr_number)
 
-    # Cache behavior: note that skip cache controls read behavior while
-    # rebuild_cache controls write behavior. The correct global behavior can
-    # be understood globally from the keyword used since rebuilding the cache
-    # requires also skipping cache lookup a user skipping the cache may not
-    # want their change to update the shared binary cache.
+    # Cache behavior: "rebuild" is no longer supported. Only 'skip'.
```

Comment:
The build cache is still disabled (and has been scrubbed from docs) but the updated cache mechanism will be massively more simple mostly by limiting scope and simplifying the cache packaging mechanism.

The upshot is that cross-pr-contamination won't be possible and the "rebuild" option is not needed. This results in additional simplification of config since other unrelated config options don't have to consider if they will "pollute" the cache.